src/explore-education-statistics-common/src/components/ContentSectionIndex.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import findAllForId from '@common/lib/dom/findAllForId';
+
+interface IndexResult {
+  element: Element;
+  location: string;
+}
+
+interface Props {
+  fromId: string;
+}
+
+interface State {
+  indexResults: IndexResult[];
+  indexComplete: boolean;
+}
+
+class ContentSectionIndex extends Component<Props> {
+  public state: State = {
+    indexResults: [],
+    indexComplete: false,
+  };
+
+  public componentDidMount() {
+    const { fromId } = this.props;
+    this.index(fromId);
+  }
+
+  private static getLocationText(element: HTMLElement): string {
+    const location: string[] = [];
+
+    element.tagName.toLowerCase();
+    location.unshift(element.textContent || '');
+    return location.join(' > ');
+  }
+
+  private index = (fromId: string) => {
+    const elements = findAllForId(fromId, 'h2,h3,h4');
+
+    const indexResults =
+      elements != null
+        ? elements.map(element => {
+            const location = ContentSectionIndex.getLocationText(element);
+
+            return {
+              element,
+              location,
+            };
+          })
+        : [];
+
+    this.setState({ indexResults, indexComplete: true });
+  };
+
+  public render() {
+    const { indexResults, indexComplete } = this.state;
+
+    return (
+      <>
+        {indexComplete && <h3 className="govuk-heading-s">In this section</h3>}
+
+        <ul className="govuk-body-s">
+          {indexComplete &&
+            indexResults.map((item, index) => {
+              return (
+                <>
+                  {item.location && (
+                    <li>
+                      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
+                      <a
+                        onClick={() => {
```

Comment:
This is both an accessibility no-no (hence all the lints you're having to disable) and will also not work in non-Javascript land. 

We need the links to have standard `href`s that reference the specific element IDs on the page e.g. `<a href="#my-element">`

Reply:
I think we'll probably be able to change this once the content editor is done & we can assign ids to things but for now this is just imported html content.